poh-bundle-win/oh-api.bat

```diff
@@ -0,0 +1,71 @@
+@echo off
+set OH_PATH=%~dps0
+set XCHANGE32_PATH=%OH_PATH%\mysql\bin
+set freePort=
+set startPort=3306
+
+:SEARCHPORT
+netstat -o -n -a | find "LISTENING" | find ":%startPort% " > NUL
+if "%ERRORLEVEL%" equ "0" (
+  echo "port unavailable %startPort%"
+  set /a startPort +=1
+  GOTO :SEARCHPORT
+) ELSE (
+  echo "port available %startPort%"
+  set freePort=%startPort%
+  GOTO :FOUNDPORT
+)
+
+:FOUNDPORT
+echo "MySQL will listen on the free port %freePort%"
+
+cd /d %OH_PATH%\oh\rsc
+echo f | xcopy dicom.properties.ori dicom.properties /y
+%XCHANGE32_PATH%\Xchang32.exe dicom.properties "OH_PATH_SUBSTITUTE" "%OH_PATH%"
+%XCHANGE32_PATH%\Xchang32.exe dicom.properties "^x5c" "^x2f"
+echo f | xcopy database.properties.sample database.properties /y
+%XCHANGE32_PATH%\Xchang32.exe database.properties "3306" "%freePort%"
+%XCHANGE32_PATH%\Xchang32.exe database.properties "^x5c" "^x2f"
+echo f | xcopy log4j.properties.ori log4j.properties /y
+%XCHANGE32_PATH%\Xchang32.exe log4j.properties "3306" "%freePort%"
+%XCHANGE32_PATH%\Xchang32.exe log4j.properties "^x5c" "^x2f
+
+cd /d %OH_PATH%\mysql\bin
+echo f | xcopy my.ori my.cnf /y
+%XCHANGE32_PATH%\Xchang32.exe my.cnf "3306" "%freePort%"
+%XCHANGE32_PATH%\Xchang32.exe my.cnf "OH_PATH_SUBSTITUTE" "%OH_PATH%"
+%XCHANGE32_PATH%\Xchang32.exe my.cnf "^x5c" "^x2f"
+
+start /b /min %OH_PATH%mysql\bin\mysqld --defaults-file=%OH_PATH%mysql\bin\my.cnf --standalone --console
```

Comment:
> Let's think about use cases: who's using the GUI and who's using the web UI?

The use case is providing both and in an easy way: altough UI is at the moment under development the distribution 2.0 (for which this PR is for) still will not have a fully featured UI from the first release, hence the GUI will be still much used and UI will be useful for both dev and end-user tests.

It would be nice to allow end user to use both GUI and UI and to be guided in how to do it: you may want to use the new UI only for patients search&registration (next MVP from UI roadmap) and still use the GUI for other features.

So we could think two different scripts where we can start the GUI and/or the UI, the important thing here, is that the embedded MySQL Server should not be started twice and that the scripts do everything "expected" without user intervention (most of the end users have no idea about what a command line is). This of course could be achieved with a cascade of scripts calling each other with proper flags.

IMHO to provide a unique script that starts GUI and also the UI (always), with a linkfile pointing to the URL is the easiest thing in terms of script complexity and end-user usability, at least at this stage.

Reply:
Personally, as an end user, I'd prefer not to have another server started locally (possibly without me even knowing it); and there could also be concurrency issues due to having GUI and Web UI applications writing to the same database.
It might be worth discussing pros and cons on a JIRA discussion page and talk about this in the next call. Anyway, this PR looks good to me (the script can be improved later).

Reply:
I personally found that any software of this kind starts a server for the DB and a server for the UI, this is quite acceptable IMHO in order to run the system and allow to do what is requested for, moreover it would be documented and the user informed.

Concurrency should be handled by Spring Frameworks in both cases (UI and GUI) thanks to LOCK feature on the records (versioning, it was in place even before Spring adoption). 

Both aspects are very important indeed, thanks to have pointed them out! ;)

Reply:
My intention here was that oh-api.bat file will only run on one machine which will act as server. Users should not be even bothered with oh-api.bat. To access the UI all they need is the browser and http link. 

Currently, if all end users run the oh.bat file, they will still connect to the same database, right ?

Reply:
> My intention here was that oh-api.bat file will only run on one machine which will act as server. Users should not be even bothered with oh-api.bat. To access the UI all they need is the browser and http link.

You are right paurav-munshi, nevertheless the portable versions (the ones we release) are meant for running on a single machine; other clients, in order to connects to the embedded MySQL server, require that the former is running all the time (and just launch gui or use the browser as you said); in a network configuration, instead, the MySQL server will be stand-alone and the only server component will be the "api", clients will just run the "gui" using the "core" as library AND/OR the browser.

So we can think about two scripts: 
- "oh.bat" in portables runs everything, both MySQL and "api" server
- FULL_DIR (also created by the build_poh.sh) "oh-server.cmd" starts only the "api" to allow clients to use browser, while "startup.cmd" reamain as is and would start only the "gui"

> Currently, if all end users run the oh.bat file, they will still connect to the same database, right ?

yes, if you are on the same machine where the embedded MySQL server, otherwise on a different machine you have to point to the running server by changing the parameters into database.properties files (I don't know if this step was already clear)

I think that much of confusion is due to the fact that we are in a transition phase where both "gui" and "ui" are meant to be used